teps/0086-changing-the-way-result-parameters-are-stored.md

```diff
@@ -0,0 +1,457 @@
+
+1. What is the default storage mechanism shipped? We want to provide a mechanism that does not require storage or additional dependencies. Configmaps is the ideal choice here, even if it creates additional ServiceAccount changes as when it comes time to production, these can be tightened as well as an alternative Results backing mechansim chosen.
+
+2. Are we wanting a centralized or distributed design? If we combine the Result Sidecar with the Dedicated HTTP Service we potentially get the best of both worlds. A centralized controller for security and extensibility. With a sidecar of reading and processing the results.
+
+### Open Design Questions
+
+- Do we need the extra byRef boolean in the model?
+- Should the sidecar be responsible for deciding whether the result should be reported by-value or by-reference? Or is that a controller-wide configuration? Passing by-value is still useful for small pieces of data to be able to have them inlined in TaskRun/PipelineRun statuses.
+- How should the sidecar report that results-writing or param-getting failed, and how should the TaskRun controller be notified so that the TaskRun can also be failed?
+
+## Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy.  Anything
+that would count as tricky in the implementation and anything particularly
+challenging to test should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations).
+-->
+
+## Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
+-->
+
+## Drawbacks
+
+<!--
+Why should this TEP _not_ be implemented?
+-->
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out?  These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+### Result References
+
+Beyond size limits of specific TaskRuns’ results, the fundamental issue is that API objects in etcd are not suitable for storing as much data as users want to be able to report in results, and pass as parameters.
+
+In addition to disaggregating TaskRun statuses into PipelineRun statuses (which we should do anyway before v1), we should introduce some way to pass one TaskRun’s output results into another TaskRun’s input parameters by reference, instead of only by value as they are today.
+
+**Example:**
+
+```yaml
+tasks:
+- name: first-task
+  taskSpec: foo
+  results:
+  - name: out
+    byRef: true # <-- *new field*
```

Comment:
I'm thinking that when we nail down the design, it would make sense for this to be entirely handled by controller configuration and not something that needs to be made explicit at authoring time. Happy to hammer this out in subsequent conversations tho.

Reply:
Added